src/api/products.ts

```diff
@@ -0,0 +1,30 @@
+import { getApi } from "./axios";
+
+export interface ProductsResponse {
+  products: Product[];
+  total: number;
+  skip: number;
+  limit: number;
+}
+
+export interface Product {
+  id: number;
+  title: string;
+  description: string;
+  price: number;
+  discountPercentage: number;
+  rating: number;
+  stock: number;
+  brand: string;
+  category: string;
+  thumbnail: string;
+  images: string[];
+}
+
+export const getProductsQuery = () => ({
+  queryKey: ["getProductsQuery", { domain: "product" }],
```

Comment:
We can remove the `{ domain: "product" }` now, and just leave `"getProductsQuery"`.